### DISCUSSION TITLE
Kovan DAI contract address

### DISCUSSION THREAD

**Discussion by delphee** [smartcontractkit/full-blockchain-solidity-course-py#569] [Q&A] [answered]
While experimenting with the AAVE testnet (staging.aave.com) I tried adding DAI to my metamask wallet... but you can't find the Kovan DAI contract address on AAVE (at least I couldn't).  How do you find the genuine Kovan DAI contract address to add to Metamask?  There seem to be many fakes out there.

**Comment by cromewar** [comment, marked as answer]
Hello @delphee , here you have: https://kovan.etherscan.io/token/0x4f96fe3b7a6cf9725f59d353f723c1bdb64ca6aa
Best regards, Cromewar.

**Reply by delphee** [reply]
cromewar, thank you for your answer.  I would like to know how I can find this on my own.  I googled it, and I used search in etherscan, but it seems there are several other DAI tokens out there that are different altogether.  Is there somewhere that lists the "genuine" testnet tokens?
So I added this token to my metamask wallet, but the DAI I borrowed on AAVE testnet  still does  not show up in my wallet.  This is why I figured I must have had the wrong DAI contract address.  What could be wrong?  I hope this isn't too far off topic.  I think it is relevant for the Defi chapters.

**Reply by delphee** [reply]
Ok, here is what I've learned:  In Metamask, with my kovan account selected, if you click on the three vertical dots next to your account address, you see the option to "View Account on Etherscan".  Click that, and once on kovan etherscan, there is a dropdown where you can see the tokens in your wallet (even if they are not visible from the Metamask app).  If you click on those tokens, you are taken to their respective contract pages, where you can see the contract address.  Copy that, add it to your Metamask wallet, and then the tokens show up.

**Reply by cromewar** [reply]
Well this is actually a tricky question.
I just sent you a DAI token with a bunch of holders, as we are on testnets there is not something like a  "genuine" token, everyone can create and publish a token that is one thing that makes blockchain great. But of course some services or platforms will prefer a specific token in that case, the best place to go is the documentation of that project. So according  to [AAVE documentation](https://docs.aave.com/developers/v/1.0/deployed-contracts/deployed-contract-instances) the contract address they use for DAI in kovan network is: https://kovan.etherscan.io/address/0xFf795577d9AC8bD7D90Ee22b6C1703490b6512FD

Give it a try,

Regarding to the mainnet, if you want to be sure the token is a trustful one you should go to pages like coinmarket cap or coin gecko which has some filters before listing any token.

Best, Cromewar.

**Reply by cromewar** [reply]
> Ok, here is what I've learned: In Metamask, with my kovan account selected, if you click on the three vertical dots next to your account address, you see the option to "View Account on Etherscan". Click that, and once on kovan etherscan, there is a dropdown where you can see the tokens in your wallet (even if they are not visible from the Metamask app). If you click on those tokens, you are taken to their respective contract pages, where you can see the contract address. Copy that, add it to your Metamask wallet, and then the tokens show up.

Nice catch!!

**Reply by delphee** [reply]
Thanks again, Cromewar!  I looked and looked for that info in the AAVE docs, but couldn't find it.
